src/TestUtils/src/DeviceTests/AssertionExtensions.iOS.cs

```diff
@@ -528,5 +529,50 @@ public static UIView GetAccessiblePlatformView(this UIView platformView)
 
 			return platformView;
 		}
+
+		public static void TapBackButton(this UINavigationBar uINavigationBar)
+		{
+			var item = uINavigationBar.FindDescendantView<UIView>(result =>
+			{
+				return result.Class.Name?.Contains("UIButtonBarButton", StringComparison.OrdinalIgnoreCase) == true;
+			});
+
+			_ = item ?? throw new Exception("Unable to locate back button view");
+
+			var recognizer = item!.GestureRecognizers!.OfType<UITapGestureRecognizer>().FirstOrDefault();
+			_ = recognizer ?? throw new Exception("Unable to Back Button TapGestureRecognizer");
+
+			recognizer.State = UIGestureRecognizerState.Ended;
+		}
```

Comment:
Is the purpose of this "TapBackButton" to try to GET the back button or to just TAP the back button? If the latter, is this state setting on line 545 all we need to simulate the tap?

Reply:
It's just tapping the back button. It's simulating a user action

Reply:
gotcha, thanks for clarifying!